src/components/CanvasPainter.jsx

```diff
@@ -78,6 +79,24 @@ const CanvasPainter = ({
       customAxis &&
       Math.abs(currentRotationAngleRef.current - rotationAngle) <= 0.01
     ) {
+      const minDistance = 0.5;
+      const maxContact = 3;
+
+      if (
+        collisionCheck(
+          rotateGroupRef,
+          nonRotateGroupRef,
+          minDistance,
+          maxContact
+        )
```

Comment:
여기서 조건으로 사용된 `collisionCheck` 함수는 `useFrame` 로직 아래에 있는 것으로 보입니다. 이런 방식은 지양하는 게 좋습니다. 

1. 자바스크립트는 일반적으로 코드가 위에서 아래로 읽히는데, 이렇게 함수를 사용한 곳보다 함수가 아래에 있다면 코드의 흐름을 이해하기 어려울 수 있습니다. 
2. `const`로 선언된 함수 표현식은 TDZ의 영향을 받습니다. 때문에 선언 전에 호출하게 된다면 `ReferenceError`가 발생할 수 있습니다.
[Temporal dead zone (TDZ)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let#temporal_dead_zone_tdz)
3. 조건문 내부에서 함수 호출이 바로 이루어질 경우 이 조건을 해석하기 위해 함수를 읽어야 한다는 단점이 있습니다. 이것은 코드를 이해하기 어렵게 만듭니다.

따라서 `collisionCheck`를 `useFrame` 위로 올리고 조건을 따로 분리해 변수에 담아 사용하는 게 좋을 것 같습니다.
```suggestion
        const collisionCheckResult = collisionCheck(
          rotateGroupRef,
          nonRotateGroupRef,
          minDistance,
          maxContact
        )
        
        if (collisionCheckResult) { ... }
```

Reply:
말씀주신 내용으로 다시 읽어보니 놓쳤던 부분 입니다. 감사합니다!